File: src/Routing/Graph/GraphAttributes/Enums.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Simple.Routing.Graph
{
    enum RoadType : sbyte
    {
        motorway = 1,
        motorway_link = 2,
        trunk = 3,
        trunk_link = 4,
        primary = 5,
        primary_link = 6,
        secondary = 7,
        secondary_link = 8,
        tertiary = 9,
        tertiary_link = 10,
        residential = 11,
        living_street = 12,
        unclassified = 13,
        road = 14,
        track = 15,
    }

    [Flags]
    enum Access
    {
        privat,
        
    }

    [Flags]
    enum Restrictions
    {
        none = 0,
        forward = 1,
        backward = 2,
        both = 4,
        noovertacking = 8,
        giveway = 16,
        stop = 32,
        narrow = 64,
    }

    enum Obstacles
    {
        hump,
        bump,
        table,
        cushion,
        dip,
        chicane,
        island,
        choker,
        crossing,
        trafficsignal,
    }

    enum Direction : byte
    {
        backward = 0,
        forward = 1,
    }
}

File: src/go-routing/graph/enums.go
package graph

//*******************************************
// enums
//*******************************************

type Direction byte

const (
	BACKWARD Direction = 0
	FORWARD  Direction = 1
)

type RoadType int8

const (
	MOTORWAY       RoadType = 1
	MOTORWAY_LINK  RoadType = 2
	TRUNK          RoadType = 3
	TRUNK_LINK     RoadType = 4
	PRIMARY        RoadType = 5
	PRIMARY_LINK   RoadType = 6
	SECONDARY      RoadType = 7
	SECONDARY_LINK RoadType = 8
	TERTIARY       RoadType = 9
	TERTIARY_LINK  RoadType = 10
	RESIDENTIAL    RoadType = 11
	LIVING_STREET  RoadType = 12
	UNCLASSIFIED   RoadType = 13
	ROAD           RoadType = 14
	TRACK          RoadType = 15
)

type Adjacency byte

const (
	ADJACENT_EDGES     Adjacency = 0
	ADJACENT_SHORTCUTS Adjacency = 1
	ADJACENT_ALL       Adjacency = 2
	ADJACENT_SKIP      Adjacency = 3
	ADJACENT_UPWARDS   Adjacency = 4
	ADJACENT_DOWNWARDS Adjacency = 5
)
